level1/lab1-kernel-thread/src/thread.c

```diff
@@ -46,34 +47,48 @@ int create_thread(uintptr_t* tid, start_routine func, void* args) {
                 | CLONE_SIGHAND | CLONE_THREAD
                 | 0);
 
-    int ret = clone(start_thread,
-                    (unsigned char*)th->st->stack_ptr + th->st->stack_size,
-                    clone_flags,
-                    (void*)th);
+    clone(start_thread,
+            (unsigned char*)th->st->stack_ptr + th->st->stack_size,
+            clone_flags,
+            (void*)th);
 
     if (errno != 0) {
```

Comment:
Here you have a situation of undefined behavior 2 lines below. Please explain how it might happen and fix it.

Reply:
where is explanation ?